libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -51,9 +121,22 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
-	for (auto const& variable: _varDecl.variables)
+	int varIndex = numVariables - 1;
+	solAssert(m_allowStackOpt || m_unusedStackSlots.empty(), "");
+	for (; varIndex >= 0 && !m_unusedStackSlots.empty(); --varIndex)
 	{
-		auto& var = boost::get<Scope::Variable>(m_scope->identifiers.at(variable.name));
+		int slot = *m_unusedStackSlots.upper_bound(0);
+		m_unusedStackSlots.erase(slot);
+		auto& var = boost::get<Scope::Variable>(m_scope->identifiers.at(_varDecl.variables[varIndex].name));
```

Comment:
It probably makes sense to assign `_varDecl.variables[varIndex]` to a variable (also used below).